### ISSUE TITLE
Have you considered to push this mainline?

### ISSUE THREAD

**Issue by XVilka** [draperlaboratory/cbat_tools#7]
Thus many BAP users would see it out of the box. And probably more testing and contributing.

**Comment by ccasin** [comment]
We'd definitely like to, but there are a couple low-hanging ways to improve it that I've been meaning to get around to first.  It's been a busy few months, but hopefully soon!

**Comment by ivg** [comment]
Well, it is already sort of pushed, see the [opam package][1]. Once it is installed, any bap user can benefit of it.


[1]: https://opam.ocaml.org/packages/cbat-tools/

**Comment by XVilka** [comment]
@ivg yes, it is available once you know about its existence. It is not mentioned anywhere on the BAP site as far as I remember. Having it under the BAP organization will provide better visibility.

**Comment by ivg** [comment]
We can include it in the bap package, this was the plan, actually. We're waiting for the first stable release of cbat_tools. Of course, cbat developers are welcome to BinaryAnalysisPlatform repo, and if they want we can grant them all the necessary rights, but I believe they will feel more comfortable under their own organization :) We are also happy to include cbat in our README.md, but this is not extensible solution. The best way to learn about the existence of this package, is to do `opam search bap`

**Comment by XVilka** [comment]
Note, now with BAP 2.0 being in master, it might require some changes in the code:

- https://github.com/BinaryAnalysisPlatform/bap/pull/960
- http://binaryanalysisplatform.github.io/knowledge-intro-1

**Comment by ccasin** [comment]
Indeed - we plan to update to BAP 2.0, but it may be a few weeks.  In general we are still happy for any of these tools to eventually be included with BAP - in particular the WP analysis is at this point quite useful.  However, since it is also still under active development, I'm not sure what the best strategy for that would be.
